src/ray/common/common_protocol.h

```diff
@@ -90,7 +90,7 @@ flatbuffers::Offset<flatbuffers::String> to_flatbuf(flatbuffers::FlatBufferBuild
 
 template <typename ID>
 ID from_flatbuf(const flatbuffers::String &string) {
-  RAY_CHECK(string.size() == ID::Size());
+  RAY_CHECK(string.size() == ID::Size()) << "expected size is " << ID::Size() << ", but got " << string.size();
```

Comment:
Can you move this `RAY_CHECK` inside `ID::FromBinary(...)`? I think that is a more appropriate place for the check.